tests/spec/core/dfn-spec.js

```diff
@@ -329,4 +329,218 @@ describe("Core — Definitions", () => {
       expect(dfnErrors).toHaveSize(2);
     });
   });
+
+  describe("CSS class based definitions types", () => {
+    it("allows exporting definitions through a class", async () => {
+      const body = `
+      <section>
+        <h2>abstract operations</h2>
+        <p>
+          <dfn class="export">just a regular definition</dfn>
+        </p>
+      </section>
+      `;
+      const ops = makeStandardOps(null, body);
+      const doc = await makeRSDoc(ops);
+      const dfn = doc.querySelector("dfn");
+      expect(dfn.textContent).toBe("just a regular definition");
+      expect(dfn.dataset.export).toBe("");
+      expect(dfn.dataset.noexport).toBeUndefined();
+
+      // Check validation error
+      const errors = doc.respec.errors.filter(err => err.plugin === "core/dfn");
+      expect(errors).toHaveSize(0);
+    });
+
+    it("handles abstract operations", async () => {
+      const body = `
+      <section>
+        <h2>abstract operations</h2>
+        <p id="abstract-op">
+          <dfn class="abstract-op">abstract operation</dfn>
+        </p>
+      </section>
+      `;
+      const ops = makeStandardOps(null, body);
+      const doc = await makeRSDoc(ops);
+      const dfn = doc.querySelector("#abstract-op dfn");
+      expect(dfn.dataset.dfnType).toBe("abstract-op");
+      expect(dfn.dataset.export).toBe("");
+    });
+
+    it("handles attributes", async () => {
+      const body = `
+        <section>
+          <h2>Attributes</h2>
+          <p id="attributes">
+            <dfn class="attribute">some-attribute</dfn>
+          </p>
+          <p id="attribute-bad">
+            <dfn class="attribute">-attribute</dfn>
+          </p>
+        </section>
+      `;
+      const ops = makeStandardOps(null, body);
+      const doc = await makeRSDoc(ops);
+      const dfn = doc.querySelector("#attributes dfn");
+      expect(dfn.dataset.dfnType).toBe("attribute");
+      expect(dfn.dataset.export).toBe("");
+
+      // Check validation error
+      const errors = doc.respec.errors.filter(err => err.plugin === "core/dfn");
+      expect(errors).toHaveSize(1);
+    });
+
+    it("handles attribute values", async () => {
+      const body = `
+        <section>
+          <h2>Attribute values</h2>
+          <p id="attr-value">
+            <dfn class="attr-value" data-dfn-for="input/type">password</dfn>
+          </p>
+          <p id="attr-value-errors">
+            <dfn class="attr-value">some-text</dfn>
+            <dfn class="attr-value">-not-ok!</dfn>
+          </p>
+        </section>
+      `;
+      const ops = makeStandardOps(null, body);
+      const doc = await makeRSDoc(ops);
+      const dfn = doc.querySelector("#attr-value dfn");
+      expect(dfn.dataset.dfnFor).toBe("input/type");
+      expect(dfn.dataset.dfnType).toBe("attr-value");
+      expect(dfn.dataset.export).toBe("");
+
+      // Check validation error
+      const errors = doc.respec.errors.filter(err => err.plugin === "core/dfn");
```

Comment:
Can move this into a small helper function in nearest `describe()` block, as it's repeated.